mass_mailing_subscription_email/controllers/main.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2022 Camptocamp SA (https://www.camptocamp.com).
+# @author Iván Todorovich <ivan.todorovich@camptocamp.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import http
+from odoo.exceptions import AccessDenied
+from odoo.http import request
+from odoo.tools import consteq
+
+from odoo.addons.mass_mailing.controllers.main import MassMailController
+
+
+class MassMailSubscriptionEmailController(MassMailController):
+    @http.route(
+        "/mail/mailing/contact/<int:res_id>/unsubscribe",
+        type="http",
+        website=True,
+        auth="public",
+    )
+    def mailing_contact_unsubscribe(self, res_id, email=None, token="", **post):
+        subscription = request.env["mailing.contact.subscription"].sudo().browse(res_id)
+        if not subscription.exists():
+            return request.redirect("/web")
```

Comment:
404 instead?

Reply:
Trying to mimic core here: https://github.com/odoo/odoo/blob/5c9b42fa1714bc9968f524bdced432b536e43042/addons/mass_mailing/controllers/main.py#L82